src/ProjectTemplates/Web.ProjectTemplates/content/WebApiAot-CSharp/Program.Main.cs

```diff
@@ -24,11 +24,24 @@ public static void Main(string[] args)
         };
 
         var todosApi = app.MapGroup("/todos");
-        todosApi.MapGet("/", () => sampleTodos);
+        todosApi.MapGet("/", () => sampleTodos)
+                #if (EnableOpenAPI)
+                .WithName("GetTodos");
+                #elif
+                ;
+                #endif
+
         todosApi.MapGet("/{id}", (int id) =>
             sampleTodos.FirstOrDefault(a => a.Id == id) is { } todo
                 ? Results.Ok(todo)
-                : Results.NotFound());
+                : Results.NotFound())
+            #if (EnableOpenAPI)
+            .WithName("GetTodoById")
+            .Produces<Todo>(StatusCodes.Status200OK)
```

Comment:
I think instead of using the `Produces` extension methods in the templates we should cosndier using the `TypedResults` extension methods and `IResult<>` return type to demonstrate these types of union-based returns.

Reply:
Done. We're now using TypedResults. I've chosen to keep using TypedResults even when the user disables OpenAPI:

- Using Results/TypedResults based on the fact if the user is using OpenAPI or not might cause confusion
- TypedResults are always better than Results, because of the added metadata and compile-time safety

Let me know if you agree, or if you want me to return to using `Results` in openapi-disabled scenarios.